src/dataExplorer/components/TagSelector.tsx

```diff
@@ -0,0 +1,172 @@
+      <div
+        className="tag-selector-value--list-item"
+        key={value}
+        onClick={() => onSelectTagValue(value)}
+      >
+        {value}
+      </div>
+    ))
+  }
+
+  const handleLoadMore = () => {
+    const newIndex = index + LOCAL_LIMIT
+    if (loadMore) {
+      // Add more key values to show
+      const newValuesToShow = concat(
+        valuesToShow,
+        slice(tagValues, index, newIndex)
+      )
+      setValuesToShow(newValuesToShow)
+    }
+    const newLoadMore = newIndex < tagValues.length
+    setLoadMore(newLoadMore)
+    setIndex(newIndex)
+  }
+
+  const loadMoreButton = loadMore && (
+    <div className="load-more-button" onClick={handleLoadMore}>
+      + Load more
+    </div>
+  )
+
+  return (
+    <Accordion className="tag-selector-value">
+      <Accordion.AccordionHeader className="tag-selector-value--header">
+        <div onClick={() => onSelectTagKey(tagKey)}>
+          <SelectorTitle title={tagKey} />
+        </div>
+      </Accordion.AccordionHeader>
+      <div className="container-side-bar">
+        {list}
+        {loadMoreButton}
+      </div>
+    </Accordion>
+  )
+}
+
+const TagSelector: FC = () => {
+  const {tags, loadingTagKeys, loadingTagValues, selectTagKey} = useContext(
+    NewDataExplorerContext
+  )
+
+  const tagKeys: string[] = Object.keys(tags)
+
+  const handleSelectTagKey = (key: string) => {
+    if (tags[key].length === 0) {
+      // Only need to get tag values if currently no values
+      selectTagKey(key)
+    }
+  }
+
+  const handleSelectTagValue = (value: string) => {
+    // TODO
+    /* eslint-disable no-console */
+    console.log(value)
+    /* eslint-disable no-console */
+  }
+
+  let list: JSX.Element | JSX.Element[] = (
+    <div className="tag-selector-key--list-item">No Tags Found</div>
+  )
+
+  if (loadingTagKeys === RemoteDataState.Error) {
+    list = (
+      <div className="tag-selector-key--list-item">Failed to load tags</div>
+    )
+  } else if (
+    loadingTagKeys === RemoteDataState.Loading ||
+    loadingTagKeys === RemoteDataState.NotStarted
+  ) {
+    list = <WaitingText text="Loading tags" />
+  } else if (loadingTagKeys === RemoteDataState.Done && tagKeys.length) {
+    list = tagKeys.map(key => {
+      return (
+        <div key={key} className="tag-selector-key--list-item">
+          <TagValues
+            tagKey={key}
+            tagValues={tags[key]}
+            onSelectTagKey={handleSelectTagKey}
+            onSelectTagValue={handleSelectTagValue}
+            loading={loadingTagValues[key]}
+          />
+        </div>
+      )
+    })
+  }
+
+  return (
+    <Accordion className="tag-selector-key" expanded={true}>
+      <Accordion.AccordionHeader className="tag-selector-key--header">
+        <SelectorTitle title="Tag Keys" info="Test info" />
+      </Accordion.AccordionHeader>
+      <div className="container-side-bar">{list}</div>
+    </Accordion>
+  )
+}
+
+export default TagSelector
```

Comment:
similar points here as about the `FieldSelector` w/r/t reducing the usage of stateful logic here in favor of deriving properties